### ISSUE TITLE
Not able to drain node as pdb is not mutable

### ISSUE THREAD

**Issue by PrincePranav235** [istio/istio#53909]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

I was trying to scale min available to 0 but didn't succeed, How can i change the pdb minavailable value to 0, why it is hard coded or it's mutable there should be flexibilty.

### Version

```Text
tag: 18.2.0
```

**Comment by Stevenjin8** [comment]
@PrincePranav235 I'm not sure what exactly your question is.

The pod disruption budget of what? also what version of kubernetes and istio are you using? What have you tried to scale the minimum available pods?

**Comment by Stevenjin8** [comment]
Ok after doing some more research I'm guessing that you mean the PodDisruptionBudget for istiod? We hardcoded it a really long time ago in https://github.com/istio/installer/pull/142.

One thing you could try is to increase the replica count of the istiod deployment to get around the pdb. You can either do this at install time or with `kubectl edit`. Let me know if you need more guidance.

**Comment by ilrudie** [comment]
+1 for scaling as a way to move past the PDB stopping a node drain if you are talking about istiod's PDB. It is also not really immutable. It's not an install-time configuration but you can change it in the cluster in a pinch.
```shell
# scale deployment
kubectl scale deployment -n istio-system istiod --replicas 2

# or change pdb
kubectl patch poddisruptionbudgets.policy -n istio-system --patch='{"spec":{"minAvailable": 0}}'
```

**Comment by PrincePranav235** [comment]
Hi @Stevenjin8 ,
yesI'am querying about pdb only, can't i set the min available in pdb to zero at install time?

**Comment by PrincePranav235** [comment]
> +1 for scaling as a way to move past the PDB stopping a node drain if you are talking about istiod's PDB. It is also not really immutable. It's not an install-time configuration but you can change it in the cluster in a pinch.
> 
> ```shell
> # scale deployment
> kubectl scale deployment -n istio-system istiod --replicas 2
> 
> # or change pdb
> kubectl patch poddisruptionbudgets.policy -n istio-system --patch='{"spec":{"minAvailable": 0}}'
> ```

yes I was able to change it in the cluster, but suppose I am running an upgrade and also don't want to scale then what is the solution to make min available of pdb to 0.

**Comment by Stevenjin8** [comment]
@PrincePranav235 I didn't see that option! Thanks for pointing that out. Yes, that works (on my machine)

```bash
 istioctl install --set values.global.defaultPodDisruptionBudget.enabled=false
```

**Comment by PrincePranav235** [comment]
I just want to know that during install time, if I set this **defaultPodDisruptionBudget** as false, then there would be PDB creation or not, as from this yaml file, it should not, but when i tried PDB was there.

Please refer the below yaml file,
https://github.com/istio/installer/pull/142/files#diff-6be7e4092dc62f9dd1d1c449ed3e5150eaf7ba23bf2aa28e48b4879438384f7aL1

and share your inputs.
Thanks

**Comment by Stevenjin8** [comment]
> I just want to know that during install time, if I set this **defaultPodDisruptionBudget** as false, then there would be PDB creation or not, as from this yaml file, it should not, but when i tried PDB was there.
> 
> Please refer the below yaml file, https://github.com/istio/installer/pull/142/files#diff-6be7e4092dc62f9dd1d1c449ed3e5150eaf7ba23bf2aa28e48b4879438384f7aL1
> 
> and share your inputs. Thanks

@PrincePranav235 How did you install istio? What command are you using? It works for me.

**Comment by ilrudie** [comment]
One side note, if the PDB is already in the cluster I don't think it would be deleted so that may account for the difference being observed

**Comment by Stevenjin8** [comment]
@ilrudie I'm 99% sure it doesn't get deleted.

```
11:34:59 vm:~/k8-lib$ istioctl install --set values.global.defaultPodDisruptionBudget.enabled=false
        |\
        | \
        |  \
        |   \
      /||    \
     / ||     \
    /  ||      \
   /   ||       \
  /    ||        \
 /     ||         \
/______||__________\
____________________
  \__       _____/
     \_____/

This will install the Istio 1.24.0 profile "default" into the cluster. Proceed? (y/N) y
✔ Istio core installed ⛵️
✔ Istiod installed 🧠
✔ Ingress gateways installed 🛬
- Pruning removed resources                                                             
  Removed policy/v1, Kind=PodDisruptionBudget/istiod.istio-system.
  Removed policy/v1, Kind=PodDisruptionBudget/istio-ingressgateway.istio-system.
✔ Installation complete
```
